src/utils/resources/vm/utils/source.ts

```diff
@@ -10,6 +14,25 @@ import { TemplateBootSource } from '@kubevirt-utils/resources/template/hooks/use
 
 import { getBootDisk, getVolumes } from './selectors';
 
+export const getPVCSourceOrSourceRef = (
+  vm: V1VirtualMachine,
+): V1beta1DataVolumeSourcePVC | V1beta1DataVolumeSourceRef => {
+  const bootDisk = getBootDisk(vm);
+  const volume = getVolumes(vm)?.find((vol) => vol.name === bootDisk?.name);
+  const dataVolumeTemplate = vm?.spec?.dataVolumeTemplates?.find(
+    (dv) => dv.metadata?.name === volume?.dataVolume?.name,
+  );
+  const sourceRef = dataVolumeTemplate?.spec?.sourceRef;
+  if (sourceRef && sourceRef?.kind === DataSourceModel.kind) {
+    return sourceRef;
+  }
+
```

Comment:
do like you did in line 25:
const pvc = dataVolumeTemplate?.spec?.source?.pvc;
if (pvc) return pvc;